js/parts/Interaction.js

```diff
@@ -461,10 +462,25 @@ extend(Chart.prototype, /** @lends Chart.prototype */ {
         } else { // else, zoom in on all axes
             event.xAxis.concat(event.yAxis).forEach(function (axisData) {
                 var axis = axisData.axis,
-                    isXAxis = axis.isXAxis;
+                    axisTop = axis.top,
+                    axisBottom = axisTop + axis.height,
+                    isXAxis = axis.isXAxis,
```

Comment:
Looking for whether is it an X axis or Y axis may fail if the chart is inverted. We should probably check `axis.horiz` instead. Check with inverted chart: http://jsfiddle.net/highcharts/mLfucwgd/5/